images/plbase/plbase-install.sh

```diff
@@ -68,21 +68,44 @@ rm -rf /tmp/pgvector
 dnf -y remove postgresql16-server-devel
 dnf -y autoremove
 
-# TODO: use standard OS Python installation? The only reason we switched to Conda
-# was to support R and `rpy2`, but now that we've removed those, we might not
-# get any benefit from Conda.
-echo "setting up conda..."
+echo "setting up uv + venv..."
 cd /
-arch="$(uname -m)"
-# Pinning the Conda version so the default Python version is 3.10. Later conda versions use 3.12 as the default.
-curl -LO https://github.com/conda-forge/miniforge/releases/download/24.3.0-0/Miniforge3-Linux-${arch}.sh
-bash Miniforge3-Linux-${arch}.sh -b -p /usr/local -f
+curl -LO https://astral.sh/uv/install.sh
+env UV_INSTALL_DIR=/usr/local/bin sh /install.sh && rm /install.sh
 
-echo "installing Python packages..."
-python3 -m pip install --no-cache-dir -r /python-requirements.txt
+# /.venv/bin/python3 -> /usr/local/bin/python3 -> /usr/share/uv/python/*/bin/python3.10
+export UV_PYTHON_INSTALL_DIR=/usr/share/uv/python
+export UV_PYTHON_BIN_DIR=/usr/local/bin
+export UV_PYTHON_DOWNLOADS=manual
+export UV_PYTHON_PREFERENCE=only-managed
+export UV_COMPILE_BYTECODE=1
+
+if [[ "$(uname)" = "Linux" ]] && [[ "$(uname -m)" = "x86_64" ]]; then
+    # https://gregoryszorc.com/docs/python-build-standalone/main/running.html
+    # v3: 64-bit Intel/AMD CPUs approximately newer than Haswell (released in 2013) and Excavator (released in 2015)
+    UV_ARCH="linux-x86_64_v3-gnu"
+elif [[ "$(uname)" = "Linux" ]] && [[ "$(uname -m)" = "aarch64" ]]; then
+    UV_ARCH="linux-aarch64-gnu"
+elif [[ "$(uname)" = "Darwin" ]] && [[ "$(uname -m)" = "arm64" ]]; then
+    UV_ARCH="macos-aarch64-none"
+else
+    echo "Unsupported architecture combination" >&2
+    exit 1
+fi
+
+# Installing to a different directory is a preview feature
+uv python install --default --preview "cpython-3.10-${UV_ARCH}"
+uv venv --seed
+ls /usr/bin
+# `uv` has the incorrect path to the system's compilers in its generated data:
+# /root/.local/share/uv/python/cpython-3.10.17-linux-aarch64-gnu/lib/python3.10/_sysconfigdata__linux_aarch64-linux-gnu.py
+# These contain references to paths like `/usr/bin/aarch64-linux-gnu-gcc`, which
+# don't exist on Amazon Linux 2023. We need to override them with the correct paths.
+export CC=/usr/bin/gcc
+export CXX=/usr/bin/g++
```

Comment:
This bug is pretty trivally reproducible:

```Dockerfile
FROM amazonlinux:2023

RUN dnf install -y gcc g++ tar

RUN curl -LO https://astral.sh/uv/install.sh \
    && sh install.sh \
    && rm install.sh \
    && source $HOME/.local/bin/env \
    && uv venv --python 3.10 \
    && uv python install --default --preview 3.10 \
    && uv pip install pygraphviz==1.14
```

Building this fails with the following:

```
error: command '/usr/bin/aarch64-linux-gnu-gcc' failed: No such file or directory
```

The actual paths look something like the following:

```
aarch64-amazon-linux-c++
aarch64-amazon-linux-g++
aarch64-amazon-linux-gcc
aarch64-amazon-linux-gcc-11
aarch64-amazon-linux-gnu-pkg-config
```

@reteps do you think this is worth reporting upstream?

Reply:
AFAICT this is a `uv` problem, yeah. Its managed Python contains paths that point to the wrong compiler.

Reply:
Gotcha, I'll report it there and they can decide if this is something that's on them to fix!

Reply:
https://github.com/astral-sh/uv/issues/12901

Reply:
From discussion with Nathan, that issue / resolution isn't blocking.